src/System.Private.Xml/src/System/Xml/Core/XmlTextReaderImpl.cs

```diff
@@ -2915,20 +2915,7 @@ private void InitStreamInput(Uri baseUri, string baseUriStr, Stream stream, byte
             SetupEncoding(encoding);
 
             // eat preamble
-            ReadOnlySpan<byte> preamble = _ps.encoding.Preamble;
-            int preambleLen = preamble.Length;
-            int i;
-            for (i = 0; i < preambleLen && i < _ps.bytesUsed; i++)
-            {
-                if (_ps.bytes[i] != preamble[i])
-                {
-                    break;
-                }
-            }
-            if (i == preambleLen)
-            {
-                _ps.bytePos = preambleLen;
-            }
```

Comment:
@VSadov, this is going to be fixed by the C# compiler, right? i.e. it won't attempt to lift spans to a state machine when there's no need, even in debug (and presumably error out if there would be a need)?